browser/ai_chat/ai_chat_agent_profile_manager.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/ai_chat_agent_profile_manager.h"
+
+#include "brave/components/ai_chat/core/browser/utils.h"
+#include "brave/components/constants/brave_constants.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile_manager.h"
+
+#if !BUILDFLAG(IS_ANDROID)
+#include "chrome/browser/themes/theme_service.h"
+#include "chrome/browser/themes/theme_service_factory.h"
+#endif
+
+namespace ai_chat {
+
+namespace {
+
+#if !BUILDFLAG(IS_ANDROID)
+const SkColor kAIChatAgentProfileThemeColor = SkColorSetRGB(253, 58, 122);
+#endif
+
+const char16_t kAIChatAgentProfileName[] = u"Leo AI Content Agent";
+
+}  // namespace
+
+AIChatAgentProfileManager::AIChatAgentProfileManager(
+    ProfileManager* profile_manager)
+    : profile_manager_(profile_manager) {
+  CHECK(profile_manager_);
+  profile_manager_->GetProfileAttributesStorage().AddObserver(this);
+  profile_manager_->AddObserver(this);
+}
+
+AIChatAgentProfileManager::~AIChatAgentProfileManager() {
+  profile_manager_->RemoveObserver(this);
+  profile_manager_->GetProfileAttributesStorage().RemoveObserver(this);
+}
+
+void AIChatAgentProfileManager::OnProfileAdded(
+    const base::FilePath& profile_path) {
+  if (profile_path.BaseName().value() == brave::kAIChatAgentProfileDir) {
+    // Some actions we might want to do only if the profile is new, but we
+    // can't access the Profile here yet. Wait for ProfileManager to call
+    // OnProfileAdded(Profile*) and remember it's new.
+    is_added_profile_new_profile_ = true;
+  }
+}
+
+void AIChatAgentProfileManager::OnProfileAdded(Profile* profile) {
+  if (profile->IsAIChatAgent() && is_added_profile_new_profile_) {
```

Comment:
I think this should probably be a CHECK for `profile->IsAIChatAgent()`? There's a synchronous path from `AddProfileToStorage` -> `observer.OnProfileAdded(profile)` and if there wasn't then this bool check could possibly be unsafe in some edge case and we'd want to take a look at it. The CHECK will ensure that are notified if the upstream code changes.

Reply:
nvm, after discussing more this is ok